src/main/java/com/flightstats/hub/dao/aws/S3Verifier.java

```diff
@@ -67,17 +65,15 @@ public S3Verifier(LastContentPath lastContentPath,
                       @Named(ContentDao.SINGLE_LONG_TERM) ContentDao s3SingleContentDao,
                       S3WriteQueue s3WriteQueue,
                       Client httpClient,
-                      ZooKeeperState zooKeeperState,
-                      CuratorFramework curator,
+                      DistributedLeadershipLockManager distributedLeadershipLockManager,
                       MetricsService metricsService) {
         this.lastContentPath = lastContentPath;
         this.channelService = channelService;
         this.spokeWriteContentDao = spokeWriteContentDao;
         this.s3SingleContentDao = s3SingleContentDao;
         this.s3WriteQueue = s3WriteQueue;
         this.httpClient = httpClient;
-        this.zooKeeperState = zooKeeperState;
-        this.curator = curator;
+        this.distributedLeadershipLockManager = distributedLeadershipLockManager;
         this.metricsService = metricsService;
         this.baseTimeoutMinutes = HubProperties.getProperty("s3Verifier.baseTimeoutMinutes", 2);
```

Comment:
This class needs refactoring sometime ! Too many arguments :)

Reply:
I have a branch that does a little of that, though I think it extracts things and adds more dependencies.... (https://github.com/flightstats/hub/pull/1129/files) Maybe I'll catch up with stuff at some point and get that cleaned up a little more.

Reply:
Awesome to see all the refactoring going in to this code. Cannot wait for the day when we are ready to integrate with Kafka :)